ietf/nomcom/views.py

```diff
@@ -924,47 +939,85 @@ def view_feedback_pending(request, year):
+    if request.method == 'POST':
+        feedback_id = request.POST.get('feedback_id', None)
+        feedback = get_object_or_404(Feedback, id=feedback_id)
+        type = request.POST.get('type', None)
+        if type:
+            if type == 'unclassified':
+                feedback.type = None
+                messages.success(request, 'The selected feedback has been de-classified. Please reclassify it in the Pending emails tab.')
+            else:
+                feedback.type = FeedbackTypeName.objects.get(slug=type)
+                messages.success(request, 'The selected feedback has been reclassified.')
+            feedback.save()
+        else:
+            types = FeedbackTypeName.objects.exclude(slug__in=settings.NOMINEE_FEEDBACK_TYPES)
+            return render(request, 'nomcom/reclassify_feedback_unrelated.html',
+                              {'year': year,
+                               'nomcom': get_nomcom_by_year(year),
+                               'feedback': feedback,
+                               'types': types,
+                               'is_chair_task' : True,
+                              })
+
+    return view_or_reclassify_feedback_unrelated(request, year, reclassify=True)
+
 @role_required("Nomcom")
 @nomcom_private_key_required
-def view_feedback_unrelated(request, year):
+def view_feedback_unrelated(request, year, reclassify=False):
+    return view_or_reclassify_feedback_unrelated(request, year, reclassify=False)
+
+def view_or_reclassify_feedback_unrelated(request, year, reclassify=False):
     nomcom = get_nomcom_by_year(year)
     feedback_types = []
     for ft in FeedbackTypeName.objects.exclude(slug__in=settings.NOMINEE_FEEDBACK_TYPES):
         feedback_types.append({'ft': ft,
                                'feedback': ft.feedback_set.get_by_nomcom(nomcom)})
 
-    return render(request, 'nomcom/view_feedback_unrelated.html',
+    template = 'nomcom/reclassify_feedback_unrelated.html' if reclassify else 'nomcom/view_feedback_unrelated.html'
+    return render(request, template,
                               {'year': year,
-                               'selected': 'view_feedback',
                                'feedback_types': feedback_types,
-                               'nomcom': nomcom})
+                               'nomcom': nomcom,
+                               'is_chair_task': reclassify,
+                               })
+
+@role_required("Nomcom Chair", "Nomcom Advisor")
+@nomcom_private_key_required
+def reclassify_feedback_topic(request, year, topic_id):
+    if request.method == 'POST':
+        feedback_id = request.POST.get('feedback_id', None)
+        feedback = get_object_or_404(Feedback, id=feedback_id)
+        feedback.type = None
+        feedback.topics.clear()
+        feedback.save()
+        messages.success(request, 'The selected feedback has been de-classified. Please reclassify it in the Pending emails tab.')
```

Comment:
Do you intentionally only handle clearing the feedback type for "topic" feedback? Why not allow reclassification as for the other flavors?

Reply:
The logic here is to reclassify a feedback item within its category (nominee, topic, unrelated). e.g. Nominee feedback can be reclassified from `questio` to `obe` (the canonical use case), while keeping its association with the nominee. Or it can be set to `unclassified`, in which case we wipe the type and nominees, and send it back to the pending queue.

Topic feedback only allows type `comment`, so the reclassification options would be `comment` (leave it unchanged) or `unclassified`. If the chair wants to reclassify a topic comment, the only reasonable option would seem to be to send it back to the pending queue, where it could be attached to a different topic, or marked as read, I don't know.